### PR TITLE
SEAB-5121: Improve Colab and add MyBinder notebook launch

### PR DESCRIPTION
**Pull request by svonworl**
**Description**
This PR:
* modifies the Colab notebook "launch with" process by adding a button, improving the URL synthesis, and using the root URL that is supplied by the webservice. 
* adds similar MyBinder notebook "launch with" functionality.

This PR includes ~~Denis'~~ Kathy's jdk->jre circleci fix.
 
Looks like we test the workflow "launch with" functionality very lightly, and primarily in the smoke tests.  Currently, there's not a smoke test for notebooks.  We should make one.  I suggest we write a ticket to do so, and implement the notebook "launch with" tests then-and-there.

**Review Instructions**
Go to a public notebook page, use the "launch with" buttons to launch on Colab and MyBinder, and confirm it worked.  Note that as of the last few weeks, mybinder.org appears to be malfunctioning: either the image build process is hanging part way through (for details, open the Logs on the mybinder page), or the site is 500ing.  All indications are that the problem is on the MyBinder side, and hopefully they will resolve it soon.

**Issue**
https://ucsc-cgl.atlassian.net/browse/SEAB-5121

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that your code compiles by running `npm run build`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] If this is the first time you're submitting a PR or even if you just need a refresher, consider reviewing our [style guide](https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#pr-checklist)
- [x] Do not bypass Angular sanitization (bypassSecurityTrustHtml, etc.), or justify why you need to do so
- [x] If displaying markdown, use the `markdown-wrapper` component, which does extra sanitization
- [x] Do not use cookies, although this may change in the future
- [x] Run `npm audit` and ensure you are not introducing new vulnerabilities
- [x] Do due diligence on new 3rd party libraries, checking for CVEs
- [x] Don't allow user-uploaded images to be served from the Dockstore domain
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.
- [x] Check whether this PR disables tests. If it legitimately needs to disable a test, create a new ticket to re-enable it in a specific milestone.

### CODE REVIEW

**Comment by denis-yuen** [comment]
> This PR includes Denis' jdk->jre circleci fix.

@kathy-t for credit actually

**Comment by denis-yuen** [comment]
> Currently, there's not a smoke test for notebooks. We should make one. I suggest we write a ticket to do so, and implement the notebook "launch with" tests then-and-there.

:+1:

File: src/app/configuration.service.ts [hunk 1]
```diff
@@ -45,6 +45,8 @@ export class ConfigurationService {
     Dockstore.BD_CATALYST_SEVEN_BRIDGES_IMPORT_URL = config.bdCatalystSevenBridgesImportUrl;
     Dockstore.BD_CATALYST_TERRA_IMPORT_URL = config.bdCatalystTerraImportUrl;
     Dockstore.ELWAZI_IMPORT_URL = config.elwaziImportUrl;
+    Dockstore.COLAB_IMPORT_URL = config.colabImportUrl;
+    Dockstore.MYBINDER_IMPORT_URL = config.mybinderImportUrl;
```
Comment (denis-yuen) [inline src/app/configuration.service.ts:49]: `MYBINDER_IMPORT_URL` or `MY_BINDER_IMPORT_URL`?

File: src/app/workflow/launch-third-party/launch-third-party.component.ts [hunk 2]
```diff
@@ -194,6 +215,14 @@ export class LaunchThirdPartyComponent extends Base implements OnChanges, OnInit
     map(([hasContent, hasHttpImports]) => this.sevenBridgesTooltip(hasContent, hasHttpImports, 'Cavatica'))
   );
 
+  colabTooltip$: Observable<string> = combineLatest([this.hasContent$]).pipe(
+    map(([hasContent]) => (hasContent ? 'Run this notebook in Google Colaboratory' : 'The notebook has no content.'))
+  );
+
+  mybinderTooltip$: Observable<string> = combineLatest([this.hasContent$]).pipe(
+    map(([hasContent]) => (hasContent ? 'Run this notebook in My Binder' : 'The notebook has no content.'))
```
Comment (denis-yuen) [inline src/app/workflow/launch-third-party/launch-third-party.component.ts:223]: `My Binder` or `MyBinder`  or `mybinder`?
The docs I can find are annoyingly coy, usually referencing by domain name (mybinder.org)

https://jupytearth.org/jupyter-resources/ecosystem/binder.html or
https://github.com/search?q=repo%3Ajupyterhub%2Fmybinder.org-deploy%20mybinder&type=code

**Review by denis-yuen — CHANGES_REQUESTED** [review]
Found https://pdfs.semanticscholar.org/c043/bef741a9616d1144e0205ac21ceae881485d.pdf 
I think what's going on is that the software itself is Binder (they also call it BinderHub for fun?) and mybinder.org is the primary deployment. 

`mybinder` on its own is not a thing. 

If they're forced to capitalize, they choose `Mybinder.org: Maintaining and sustaining a public service`

**Comment by svonworl** [comment]
> Found https://pdfs.semanticscholar.org/c043/bef741a9616d1144e0205ac21ceae881485d.pdf I think what's going on is that the software itself is Binder (they also call it BinderHub for fun?) and mybinder.org is the primary deployment.
> 
> `mybinder` on its own is not a thing.
> 
> If they're forced to capitalize, they choose `Mybinder.org: Maintaining and sustaining a public service`

When I google "mybinder", I see this:
image
mybinder.org refers to itself, at least sometimes, as "MyBinder".

But I think it's a good solution to refer to `mybinder.org` in the UI, and use the term `mybinder` (with the expected capitalized variants) to name variables in our source.  I adjusted the PR accordingly.

**Comment by denis-yuen** [comment]
> When I google "mybinder", I see this:

Yeah, that's weird. And then if you go to it you only see "Binder" with the only reference to "my" being the button saying "Donate to mybinder.org"

:shrug:

File: src/app/workflow/launch-third-party/launch-third-party.component.html [hunk 3]
```diff
@@ -263,14 +263,51 @@
...
+        <div fxLayout="column" fxLayoutAlign="space-between center" fxLayoutGap="0.625rem">
+          <a
+            mat-button
+            class="launch-with-button private-btn"
+            [matTooltip]="colabTooltip$ | async"
+            matTooltipPosition="left"
+            target="_blank"
+            rel="noopener"
+            [attr.href]="
+              config.COLAB_IMPORT_URL +
```
Comment (coverbeck) [inline src/app/workflow/launch-third-party/launch-third-party.component.html:275]: If Colab and MyBinder only accept references to GitHub repos (?), then if we ever add notebooks support to BitBucket and/or Dockstore hosted, these links probably shouldn't exist or be disabled.

An extreme forwards looking edge case, as we only support GitHub today.

**Review by coverbeck — APPROVED** [review]

**Review by kathy-t — APPROVED** [review]

**Pull request merged**